### DISCUSSION TITLE
Does Hardhat modify the RPC URL?

### DISCUSSION THREAD

**Discussion by ebaizel** [NomicFoundation/hardhat#3813] [Troubleshooting]
I'm trying to debug why an RPC endpoint is not working within Hardhat, when it works for curl requests. The RPC endpoint is case sensitive, and it includes a query parameter.

Does Hardhat do something like lowercasing the url, or removing query parameters?
Is there a way to see what rpc endpoint it is calling? I've enabled `--verbose` but that doesn't display that.

**Comment by fvictorio** [comment]
> Does Hardhat do something like lowercasing the url, or removing query parameters?

I'm pretty sure we don't do that. If we do, is an error.

> Is there a way to see what rpc endpoint it is calling? I've enabled --verbose but that doesn't display that.

We don't have a way to do that. But it would make sense to add something like that to the verbose output. Can you tell me how you are using the remote endpoint? Is it a network you are connecting to, or is it a network you are forking?

**Comment by fvictorio** [comment]
Hmm, maybe this issue is related to your problem? https://github.com/NomicFoundation/hardhat/issues/3815
